### ISSUE TITLE
False negative for uninitialized variable usage in negative comprehensions

### ISSUE THREAD

**Issue by noloerino** [PyCQA/pylint#3506]
### Steps to reproduce
(moved from #3487)
```
b = [x for x in range(10) if y % 2 == 0 for y in range(x)]
```

### Current behavior
```
--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)
```

### Expected behavior
Semantically, the comprehension should be roughly equivalent to the following:
```python3
for x in range(10):
    if y % 2 == 0:
        for y in range(x):
            x
```
As such, the usage of y in the if statement is invalid, and should be reported as usage before assignment.

### pylint --version output
```
pylint 2.4.4
astroid 2.3.3
Python 3.7.6 (default, Dec 30 2019, 19:38:28) 
[Clang 11.0.0 (clang-1100.0.33.16)]
```
